src/Adapter/Product/Combination/QueryHandler/GetEditableCombinationsListHandler.php

```diff
@@ -164,20 +192,32 @@ private function formatEditableCombinationsForListing(
                 $combinationAttributesInformation,
                 (bool) $combination['default_on'],
                 $impactOnPrice,
-                (int) $this->stockAvailableRepository->getForCombination(new CombinationId($combinationId))->quantity
-// @todo:
-//      Missing combination image:
-//      Old page retrieves it through src/PrestaShopBundle/Controller/Admin/AttributeController::getFormImagesAction.
-//      we could simply get Product::getCombinationImageById and load new Image()->getbasePath,
-//      but not all combinations seems to have associated images with product
-//      (the old page still shows images for all of them - not sure if that is good behavior)
-//      also it is unclear how old page appends suffixes "small_default, home_default" (it seems controller only provides base path)
+                (int) $this->stockAvailableRepository->getForCombination(new CombinationId($combinationId))->quantity,
+                $this->getImagePath((int) $imageData['id_image'], $langIso)
             );
         }
 
         return new CombinationListForEditing($totalCombinationsCount, $combinationsForEditing);
     }
 
+    /**
+     * @param int $imageId
+     * @param string $langIso
+     *
+     * @return string
+     */
+    private function getImagePath(int $imageId, string $langIso): string
```

Comment:
not sure if this is good approach (I don't like hardcoded image types, as you know in theory they might be missing), but didn't find better way. Also, I feel like the "default image" responsibility should go to controller, so actually the dto should have the imagePath as Null|string. But I hesitated to do that to avoid having this code in controller :thinking:

Reply:
> I don't like hardcoded image types

Yes, but you did not choose it. It's related to current implementation. So it's alright your code represents the reality behind. It could `abcdzje.jpg` it would not change. It's just a name.

> Also, I feel like the "default image" responsibility should go to controller

Agreed. `getImagePath()` should return either valid image paths or `null`.

The controller chooses what to do with `null` image paths
- dont display them
- display another image, maybe a default
- add a "Missing picture, _click here_ to upload one" link

"How I render combinations without images" is a view reponsibility so it's OK to have it
- in the controller if logic is small enough
- in a app-level small service if logic is big enough
- if it's strongly related to Twig, in a Twig extension?